scilpy/tractograms/tractogram_operations.py

```diff
@@ -843,3 +854,246 @@ def split_sft_randomly_per_cluster(orig_sft, chunk_sizes, seed, thresholds):
     'concatenate': 'concatenate',
     'lazy_concatenate': 'lazy_concatenate'
 }
+
+
+def _compute_difference_for_voxel(chunk_indices,
+                                  skip_streamlines_distance=False):
+    """
+    Compute the difference between two sets of streamlines for a given voxel.
+    This function uses global variable to avoid duplicating the data for each
+    chunk of voxels.
+
+    Use the function tractogram_pairwise_comparison() as an entry point.
+
+    Parameters
+    ----------
+    chunk_indices: list
+        List of indices of the voxel to process.
+    skip_streamlines_distance: bool
+        If true, skip the computation of the distance between streamlines.
+
+    Returns
+    -------
+    results: list
+        List of the computed differences in the same order as the input voxel.
+    """
+    global sft_1, sft_2, matched_points_1, matched_points_2, tree_1, tree_2, \
+        sh_data_1, sh_data_2
+    results = []
+    for vox_ind in chunk_indices:
+        vox_ind = tuple(vox_ind)
+
+        global B
+        has_data = sh_data_1[vox_ind].any() and sh_data_2[vox_ind].any()
+        if has_data:
+            sf_1 = np.dot(sh_data_1[vox_ind], B)
+            sf_2 = np.dot(sh_data_2[vox_ind], B)
+            corr = np.corrcoef(sf_1, sf_2)[0, 1]
+        else:
+            corr = -1
```

Comment:
Is it a good choice to use -1 when the correlation does not exist? A correlation of -1 is a number that exists (perfect anti-correlation).

Should you explain in the script or somewhere the possible cases why a correlation would be -1?

Reply:
I can also put a 0? I would rather avoid NaN as they often break everything afterward (other scipts).

Do you think 0 is better?

Reply:
I would think so... But it depends on the use case I guess. Or like, -2, and people know that it's not possible? In future scripts, do you have options to reject some values?